web/src/settings.ts

```diff
@@ -40,6 +40,32 @@ function user_can_change_password(): boolean {
     return realm.realm_email_auth_enabled;
 }
 
+function user_can_change_role(): boolean {
+    const user = people.get_user_by_id_assert_valid(current_user.user_id);
+    if (user.is_owner && !people.is_current_user_only_owner()) {
+        return true;
+    } else if (user.is_admin && !user.is_owner) {
+        return true;
+    }
+    return false;
+}
```

Comment:
Why are you getting the user again? Can't we do: 
```suggestion

function user_can_change_role(): boolean {
    if (current_user.is_owner && !people.is_current_user_only_owner()) {
        return true;
    } else if (current_user.is_admin && !current_user.is_owner) {
        return true;
    }
    return false;
}
```